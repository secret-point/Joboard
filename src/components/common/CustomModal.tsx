import React, { useEffect } from "react";
import { Modal } from "@amzn/stencil-react-components/modal";
import { Col, Row } from "@amzn/stencil-react-components/layout";
import { IconCross, IconSize } from "@amzn/stencil-react-components/icons";
import { CommonColors } from "../../utils/colors";

interface CustomModalProps {
  shouldOpen: boolean;
  children: React.ReactNode;
  setShouldOpen: Function;
}
const CustomModal = (props: CustomModalProps) => {

  const { children, shouldOpen, setShouldOpen } = props;

  useEffect(() => {

  }, [shouldOpen]);

  return (
    <Modal isOpen={shouldOpen} close={close}>
      <Col
        backgroundColor="neutral0"
        padding={{ top: "S400", bottom: "S600", left: "S600" }}
        alignItems="center"
      >
        <Col alignItems="flex-end" width="100%" padding={{ right: "S400" }}>
          <Row
            onClick={() => {
              setShouldOpen(false);
            }}
            tabIndex={0}
            onKeyDown={(event) => {
              if (event.key === "Enter" || event.keyCode === 32) {
                setShouldOpen(false);
              }
            }}
          >
            <IconCross
              title="Close Modal"
              size={IconSize.Large}
              color={CommonColors.Blue70}
            />
          </Row>
        </Col>
        <Col padding={{ right: "S600" }} width="100%">
          {children}
        </Col>
      </Col>
    </Modal>
  );
};

export default CustomModal;