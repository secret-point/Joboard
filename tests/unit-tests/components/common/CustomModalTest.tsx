import React from "react";
import { shallow } from "enzyme";
import CustomModal from "../../../../src/components/common/CustomModal";

describe("Custom Modal", () => {
  it("should match spanshot with modal close", () => {
    const wrapper = shallow(
      <CustomModal shouldOpen={false} setShouldOpen={jest.fn}>
        <div>Child props</div>
      </CustomModal>);
    expect(wrapper).toMatchSnapshot();
  });

  it("should match spnashot with modal open", () => {
    const wrapper = shallow(
      <CustomModal shouldOpen setShouldOpen={jest.fn}>
        <div>Child props</div>
      </CustomModal>);
    expect(wrapper).toMatchSnapshot();
  });
});