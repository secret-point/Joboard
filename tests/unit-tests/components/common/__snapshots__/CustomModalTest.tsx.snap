// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Modal should match spanshot with modal close 1`] = `
<Modal
  close={[Function]}
  isOpen={false}
  isScrollable={true}
  shouldCloseOnClickOutside={false}
>
  <ModalContent
    maxWidth="50vw"
    titleText=""
  >
    <Col
      backgroundColor="neutral0"
      width="100%"
    >
      <Col
        alignItems="flex-end"
        width="100%"
      >
        <Row
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <IconCross
            color="#1768C9"
            display="block"
            size={32}
            title="Close Modal"
          />
        </Row>
      </Col>
      <Col
        width="100%"
      >
        <div>
          Child props
        </div>
      </Col>
    </Col>
  </ModalContent>
</Modal>
`;

exports[`Custom Modal should match spnashot with modal open 1`] = `
<Modal
  close={[Function]}
  isOpen={true}
  isScrollable={true}
  shouldCloseOnClickOutside={false}
>
  <ModalContent
    maxWidth="50vw"
    titleText=""
  >
    <Col
      backgroundColor="neutral0"
      width="100%"
    >
      <Col
        alignItems="flex-end"
        width="100%"
      >
        <Row
          onClick={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          <IconCross
            color="#1768C9"
            display="block"
            size={32}
            title="Close Modal"
          />
        </Row>
      </Col>
      <Col
        width="100%"
      >
        <div>
          Child props
        </div>
      </Col>
    </Col>
  </ModalContent>
</Modal>
`;
